FreeRTOS/FreeRTOS
# Fix crash on Ubuntu 24.04 ARM64

Description
-----------
On Ubuntu 24.04 ARM64, `PTHREAD_STACK_MIN` is `0x20000` which does not fit in `unsigned short`, size of which is 2 bytes. Casting `PTHREAD_STACK_MIN` to `unsigned short` in FreeRTOSConfig.h results in `configMINIMAL_STACK_SIZE` getting defined to 0 which leads to SIGSEV.

This PR removes the not needed casting of `PTHREAD_STACK_MIN` to `unsigned short`.

Test Steps
-----------
Build and run Posix_GCC demo on Ubuntu 24.04 ARM64.

Checklist:
----------


- [NA] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
https://forums.freertos.org/t/posix-gcc-demo-not-running-on-ubuntu-server-24-04-arm64/21233


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
